### PR TITLE
Adding applications

### PR DESCRIPTION
**Pull request by kevincasey1222**
Hi @austinkelleher !

Per https://jupiterone.atlassian.net/jira/software/c/projects/INT/issues/INT-2011 , here is code to add Applications and user/group relationships to those applications.

### CODE REVIEW

File: src/steps/applications/index.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      await client.iterateAppBoundGroups(appId, async (group) => {
+        const groupId = group.id;
+        if (!groupId) {
+          return;
+        }
+        const groupEntity = await jobState.findEntity(groupId);
+
+        if (!groupEntity) {
+          // It's possible that this group was created in between the time that
+          // we collected all groups and now. Just ignore this one and carry on!
```
Comment (aiwilliams) [inline src/steps/applications/index.ts:96]: I love these comments!

File: src/steps/users/index.ts [hunk 2]
```diff
@@ -22,6 +23,13 @@ export async function fetchUsers({
   });
 
   const orgEntity = await jobState.getData<Entity>(PRIORITY_ORG_CACHE_KEY);
+  if (!orgEntity) {
+    throw new IntegrationError({
+      message: 'Expected fetch-orgs to have set the Org Entity in the jobstate',
+      code: 'ORG_LIST_FAILED',
+      fatal: true,
+    });
+  }
```
Comment (aiwilliams) [inline src/steps/users/index.ts:32]: What do you think about making this a utility function?

File: test/recording.ts [hunk 3]
```diff
@@ -18,6 +18,16 @@ export async function withRecording(
     redactedRequestHeaders: ['x-api-key'],
     mutateEntry(entry) {
       mutations.unzipGzippedRecordingEntry(entry);
+      const responseContent = JSON.parse(entry.response.content.text);
+      if (responseContent.results && Array.isArray(responseContent.results)) {
+        for (const result of responseContent.results) {
+          if (result.config) {
+            //has SAML certs in it
+            result.config = '[REDACTED]';
```
Comment (aiwilliams) [inline test/recording.ts:26]: Awesome

**Review by aiwilliams — APPROVED** [review]

**Comment by kevincasey1222** [comment]
Looks like this branch has branch protection that requires the Node 12.x Ubuntu test. We'll need someone with the privilege to modify branch protections in the repo to go in and remove the Node 12.x Ubuntu test as mandatory, and then add the Node 14.x Ubuntu test (because SDK version 7+ doesn't support Node 12.x).

**Pull request merged**